app/models/application_record.rb

```diff
@@ -12,9 +12,16 @@ def append_url_options(path, options = {})
       path = "#{path}.#{options[:format]}"
     end
 
-    query_params = {}
-    query_params[:cachebust] = options[:cachebust] if options[:cachebust]
-    query_params[:preview] = options[:preview] if options[:preview]
+    permitted_query_params = %i[
+      cachebust
+      preview
+      token
+      utm_campaign
+      utm_medium
+      utm_source
+    ]
+
+    query_params = options.slice(*permitted_query_params).compact
```

Comment:
oh very nice! It slightly bugged me the way we were doing this before but wasn't sure what the better ruby option was